src/randoop/reflection/ReflectionManager.java

```diff
@@ -107,8 +107,28 @@ public void apply(Class<?> c) {
     
   }
 
-
-
+  /**
+   * Applies the visitors to the methods of the class. 
+   * Will visit all methods unless visiting only public members.
+   * 
+   * @param c  the class whose methods should be visited
+   */
+  private void applyMethods(Class<?> c) {
+    Set<Method> methods = new HashSet<>();
```

Comment:
This set is never side-effected, so it is always empty and the test `! methods.contains(m)` is always false.
I would find this method easier to understand if it was one loop, and that loop contained a single if test (albeit one with a slightly more elaborate test).  In particular, I couldn't tell from the class documentation how publicOnly and the predicate are combined:  is it an and-style test or an or-style test?  The class documentation could be made more explicit on this point as well.